### ISSUE TITLE
passing `debug:true` has no effect in forcing the addition of the stacktrace into errors on NODE_ENV test.

### ISSUE THREAD

**Issue by eturino** [apollographql/apollo-server#4107]
## The problem

`new ApolloServer({ …, debug: true })` should have forced enabling the stacktrace in NODE_ENV=test. It does not work though. Adding it to the ApolloServer used in tests

### The code I am using

In my mocks.ts file which I use for tests, I have this function:

```typescript
import { ApolloServerTestClient, createTestClient } from "apollo-server-testing";

//...

export function createTestApolloServer({
  dataSources = {},
  ...customContextParams
}: MockContextParams = {}): ApolloServerTestClient {
  const server = new ApolloServer({
    schema,
    dataSources: (): DataSources => mockDataSources(dataSources),
    context: (): CustomContext => mockCustomContext(customContextParams),
    debug: true,
  });

  return createTestClient(server);
}
```

the `debug: true` doesn't have any effect, and I still didn't get the exception stacktrace, until I patched apollo-server-core in node_modules (see the solutions section)

### Where seems to be the problem

I have tracked down the problem to `apollo-server-core` in the current version. 

The call to `formatApolloError` in https://github.com/apollographql/apollo-server/blob/71361880b6ba284ee8d16d5091945a999b005703/packages/apollo-server-core/src/requestPipeline.ts#L567 uses `requestContext.debug`.

While the type accepts an optional `debug` boolean field, it is never added in the creation of requestContext https://github.com/apollographql/apollo-server/blob/master/packages/apollo-server-core/src/ApolloServer.ts#L840

The solution could be either to use the `debug` from the config, or to add the `debug` into the requestContext.

### Why I haven't opened a PR directly

I do not have enough knowledge of the roles that both config and requestContext have to make a decision on what is the best approach.

### Possible solution 1: Use the debug from config:

change this line 
https://github.com/apollographql/apollo-server/blob/71361880b6ba284ee8d16d5091945a999b005703/packages/apollo-server-core/src/requestPipeline.ts#L567 to be 
```
      debug: config.debug,
```

### Possible solution 2: Add debug to requestContext

change 
https://github.com/apollographql/apollo-server/blob/master/packages/apollo-server-core/src/ApolloServer.ts#L840

```typescript
    const requestCtx: GraphQLRequestContext = {
      logger: this.logger,
```

to 

```typescript
    const requestCtx: GraphQLRequestContext = {
      debug: options.debug,
      logger: this.logger,
```

**Comment by bwoodlt** [comment]
Having this issue... any updates on this?

**Comment by harm-less** [comment]
It seems I encountered the same issue

**Comment by ChamNouki** [comment]
Facing the same issue here ✋

**Comment by jtomaszewski** [comment]
This makes unit/integration testing app using apollo very annoying, as errors do not show app stacktraces, and one has to guess where's their error at.

I could submit a PR but also I'm not yet familiar with those configs and request contexts.

As a temporary workaround, I was thinking about having some `postinstall` hook that would change the https://github.com/apollographql/apollo-server/blob/71361880b6ba284ee8d16d5091945a999b005703/packages/apollo-server-core/src/requestPipeline.ts#L567 line, however I'm afraid if it could have any impact on our non-test environment.

**Comment by santialbo** [comment]
@jtomaszewski I recommend you have a look at https://www.npmjs.com/package/patch-package
It allows you to do the same in a more maintainable way less susceptible to errors.
I use it when having issues like this with other libraries

**Comment by jtomaszewski** [comment]
Thanks @santialbo , didn't know that.

Still, need to confirm first it won't break `debug: boolean` option that we pass to our GQL servers in dev/prod environments. ; ] Or maybe just grasp those config/context here and submit a PR.

**Comment by ibratoev** [comment]
Yeah, I am facing the same issue. Both of your solutions look like they make sense but I need to dig deeper into the code base.

**Comment by ibratoev** [comment]
I looked deeper into the codebase and it seems the second approach is safer. The `executeOperation` function is used only for such testing utilities.

**Comment by ibratoev** [comment]
After I dug a bit in the codebase, I would consider that a bug so I opened a PR with a fix.

**Comment by glasser** [comment]
I've published a release with this to version `2.21.1-alpha.0`. My plan is to release 2.21.1 on Monday. If you want to test that this fix works before then, try running the prerelease yourself, and provide feedback on https://github.com/apollographql/apollo-server/pull/4990
